x/market/types/message_deal.go

```diff
@@ -71,3 +71,43 @@ func (msg *MsgCreateDeal) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{creator}
 }
+
+var _ sdk.Msg = &MsgSellData{}
+
+func NewSellData(cert DataValidationCertificate, seller string) *MsgSellData {
+	return &MsgSellData{
+		Cert:   &cert,
+		Seller: seller,
+	}
+}
+
+func (msg *MsgSellData) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgSellData) Type() string {
+	return "SellData"
+}
+
+// ValidateBasic is validation for MsgCreateDeal.
+func (msg *MsgSellData) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Seller)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+
+	return nil
+}
+
+func (msg *MsgSellData) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgSellData) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Seller)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
```

Comment:
```suggestion
	signer, err := sdk.AccAddressFromBech32(msg.Cert.UnsignedCert.RequesterAddress
	if err != nil {
		panic(err)
	}
	return []sdk.AccAddress{signer}
```
I think signer and requesterAddress will be the same.
So if you bring a signer, how about using it like this?

Reply:
I agree with your review. The ```seller``` is duplicated with ```RequesterAddress```. Thank you! I'll change it.

Reply:
In addition, the ```GetSigners``` method is auto-generated, but I think that it would be better to change.